homeassistant/components/smhi/weather.py

```diff
@@ -112,140 +125,65 @@ def __init__(
     ) -> None:
         """Initialize the SMHI weather entity."""
 
-        self._name = name
-        self._latitude = latitude
-        self._longitude = longitude
+        self._attr_name = name
+        self._attr_unique_id = f"{latitude}, {longitude}"
         self._forecasts: list[SmhiForecast] | None = None
         self._fail_count = 0
-        self._smhi_api = Smhi(self._longitude, self._latitude, session=session)
+        self._smhi_api = Smhi(longitude, latitude, session=session)
         self._attr_device_info = DeviceInfo(
             entry_type=DeviceEntryType.SERVICE,
-            identifiers={(DOMAIN, f"{self._latitude}, {self._longitude}")},
+            identifiers={(DOMAIN, f"{latitude}, {longitude}")},
             manufacturer="SMHI",
             model="v2",
-            name=self._name,
+            name=self._attr_name,
             configuration_url="http://opendata.smhi.se/apidocs/metfcst/parameters.html",
         )
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique id."""
-        return f"{self._latitude}, {self._longitude}"
+        self._attr_condition = None
+        self._attr_temperature = None
```

Comment:
These are defaults already

```suggestion
```

Reply:
These are not set as none in the WeatherEntity so therefore need to do it here or if data is not available it will crash.
https://github.com/home-assistant/core/blob/dev/homeassistant/components/weather/__init__.py#L112

Reply:
Right, but the temperature is. Updated suggestion.

Reply:
It's not. Not sure why but several of these _attr are typed but not set to None.
Temperature: https://github.com/home-assistant/core/blob/dev/homeassistant/components/weather/__init__.py#L121

Reply:
Maybe we should do a PR for that first and then update this PR?

Reply:
No 🤦 I've just had a long day... my bad. 🙏